### PR TITLE
Refactor and speed up repeated builds

### PR DESCRIPTION
**Pull request by mariux**
- remove semaphore/ targets again as suggested by @soerenmartius 
- simplify makefile variables
- on repeated builds load container from cache
- check versions on builds
- add test to actually execute each build tool
- add NOCOLOR option ot make
- update terraform to 0.12.25
- update tflint to 0.16.0
- install explicit version of pre-commits 2.4.0

needs release bump to 0.3.0 due to update in tflint from 0.15.x to 0.16.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,54 +1,76 @@
...
 .PHONY: docker/build
 ## Build the docker image
 docker/build:
-	docker build -t ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_TAG} .
+	docker build \
+	  --build-arg TERRAFORM_VERSION=${TERRAFORM_VERSION} \
+	  --build-arg PACKER_VERSION=${PACKER_VERSION} \
+	  --build-arg TFLINT_VERSION=${TFLINT_VERSION} \
+	  --build-arg PRECOMMIT_VERSION=${PRECOMMIT_VERSION} \
+	  -t ${DOCKER_IMAGE} . | cat
```
Comment (soerenmartius) [inline Makefile:68]: for what reason did you pipe the output of each target to cat? Won't this suppress any potential error code that might occur inside the chain of commands?

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -6,33 +6,50 @@ agent:
...
+          - make docker/load
       jobs:
         - name: Run Snyk
           commands:
             - make test/snyk
+        - name: Run Tools
+          commands:
+            - make test/execute-tools
 
 promotions:
```
Comment (soerenmartius) [inline .semaphore/semaphore.yml:54]: hm, somehow, the promotion runs twice now

Screenshot 2020-05-17 at 09 59 27

File: Makefile (hunk 1 above)
Comment (mariux) [reply Makefile:68]: true... hmm... let me think about other ways...

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (mariux) [reply .semaphore/semaphore.yml:54]: started it twice to see what happens...

Comment (soerenmartius) [reply .semaphore/semaphore.yml:54]: ah gotcha!

File: Makefile (hunk 1 above)
Comment (soerenmartius) [reply Makefile:68]: you can just get rid of the cat or for what would you use it in the first place?

Comment (mariux) [reply Makefile:68]: fixed. good catch ;) thanks..

File: Makefile [hunk 3]
```diff
@@ -59,20 +81,18 @@ docker/login:
...
+	docker save ${DOCKER_IMAGE} > "${CACHE_FILE}"
 
-.PHONY: semaphore/restore
+.PHONY: docker/load
 ## Load saved image
-semaphore/restore:
-	cache restore ${DOCKER_IMAGE_TAG}
-	docker load < docker-image.tar | cat
+docker/load:
+	docker load < "${CACHE_FILE}" | cat
```
Comment (soerenmartius) [inline Makefile:95]: remove cat

File: Makefile [hunk 4]
```diff
@@ -59,20 +81,18 @@ docker/login:
 ## Push docker image to hub.docker.com
 docker/push:
 	if [[ "${DOCKER_IMAGE_TAG}" == v[0-9]* ]] ; then docker push ${DOCKER_HUB_REPO}:latest | cat ; fi
-	docker push ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_TAG} | cat
+	docker push ${DOCKER_IMAGE} | cat
```
Comment (soerenmartius) [inline Makefile:84]: you still have a bunch of cats here

**Pull request merged**